### PR TITLE
Elaborate on hook level confinement, add a few examples

### PR DESCRIPTION
**Pull request by scop**
Per discussion at and around
https://github.com/pre-commit/pre-commit/issues/2076#issuecomment-933717382

### CODE REVIEW

File: sections/advanced.md [hunk 1]
```diff
@@ -228,6 +228,14 @@ If you do not want to have hooks installed by default on the stage passed
 during a `pre-commit install --hook-type ...`, please set the [`default_stages`](#top_level-default_stages)
 top level configuration property to the desired stages, also as an array.
 
+Hooks can also confine themselves to a set of stages through their [`stages`](#hooks-stages)
```
Comment (asottile) [inline sections/advanced.md:231]: the sentence at 222 already says essentially this

Comment (scop) [reply sections/advanced.md:231]: That sentence talks about user config in `.pre-commit-config.yaml`, while this added one is about the hook "definition" config in `.pre-commit-hooks.yaml`.

Comment (asottile) [reply sections/advanced.md:231]: the settings work the same independent of where they are defined

Comment (scop) [reply sections/advanced.md]: Combined the paragraphs in e9c320aa20c6e261f648373bc285af1b62b773c9

**Review by asottile — APPROVED** [review]

**Pull request merged**